### ISSUE TITLE
Image transform performed in edit mode if wrapped in !edit block

### ISSUE THREAD

**Issue by jrondorf** [yourhead/s3#555]
**ImageFunItem.devstack** contains in `each.html` following templates:

```
%[if edit]%  
%image%
%[endif]%%[if !edit]%
<li><img src='%id=thumbnail%' data-large='%id=large%' data-large-2x='%id=large2x%' /></li>
%[endif]%
```

When adding a lot of these ImageFunItems on a Stacks page, switching back and forth between this page and another one (all in **edit** mode, I am **not** talking about preview) takes a long time for recalculating the view. 100 ImageFunItems on a Stacks page take 15-20 seconds.

When **removing** the `%[if !edit]%` code block, switching back and forth is quite fluent and fast:

```
%[if !edit]%
<li><img src='%id=thumbnail%' data-large='%id=large%' data-large-2x='%id=large2x%' /></li>
%[endif]%
```

For me, it is unclear why the content inside the `%[if !edit]%` block seems to be executed in **edit** mode.

**Comment by jrondorf** [comment]
Same is true if wrapping the image transform templates into `%[if preview]%` or `%[if publish]%`.

**Comment by yourhead** [comment]
it seems conditionals are causing cache flushes in unexpected ways.  i'm not certain that this is feasible to fix. if not then the bug here is to document the limitation.

**Comment by yourhead** [comment]
i believe this is now fixed in 3.5 b1
